src/core/speechrecognition/locales.js

```diff
@@ -0,0 +1,23 @@
+import { SPEECH_RECOGNITION_LOCALES_SUPPORTED_BY_EDGE } from '../constants';
+
+/**
+ * Transforms the given locale to a locale Microsoft Edge can understand.
+ * This means changing the language/locale separating underscore to a dash,
+ * and defaulting the locale to the 2 character language code if it is not
+ * supported by Edge.
+ *
+ * @param {string} locale
+ * @returns {string}
+ */
+export function transformSpeechRecognitionLocaleForEdge (locale) {
+  const underscoreIndex = locale.indexOf('_');
+  if (underscoreIndex === -1) {
+    return locale;
+  }
+  locale = locale.replace('_', '-');
```

Comment:
nit: modifying the locale here means this function have the side-effect of modifying the input param. It's just a string so it's probably okay, but still something to watch out for

Reply:
this won't modify the original locale though! You can test this in console with a simple function like
```js
function mutate(a) { a = '123' }
```
and passing in a string

Reply:
I think this is only an issue with mutating the data inside an object, not reassignment

Reply:
ah, you're right!